src/ol/render/canvas/TextBuilder.js

```diff
@@ -336,7 +337,10 @@ class CanvasTextBuilder extends CanvasBuilder {
         );
         if (textState.backgroundFill) {
           this.updateFillStyle(this.state, this.createFill);
+          const fillStyle = this.state.fillStyle;
+          this.state.fillStyle = defaultFillStyle;
```

Comment:
This won't create a `SET_FILL_STYLE` instruction. Calling `updateFillStyle()` will.